### PR TITLE
Remove approved tokens parameter from TxScripts

### PR DESCRIPTION
**Pull request by DevRozaDev**
\+ some minor changes that have to be reviewed in the 1st commit

### CODE REVIEW

File: contracts/storage/tick.ral [hunk 1]
```diff
@@ -50,10 +50,9 @@ Abstract Contract TickHelper() extends Decimal() {
             newLiquidity = tick.liquidityGross + liquidityDelta
         } else {
             newLiquidity = tick.liquidityGross - liquidityDelta
+            assert!(tick.liquidityGross >= liquidityDelta && newLiquidity < maxLiquidityPerTick, InvariantError.InvalidTickLiquidity)
```
Comment (Sniezka1927) [inline contracts/storage/tick.ral:53]: Are you sure it should be only in the `else` check?

**Review by Sniezka1927 — CHANGES_REQUESTED** [review]

File: contracts/storage/tick.ral (hunk 1 above)
Comment (DevRozaDev) [reply contracts/storage/tick.ral:53]: Good catch

File: src/invariant.ts [hunk 2]
```diff
@@ -253,8 +249,7 @@ export class Invariant {
           xToY,
           amount,
           byAmountIn,
-          sqrtPriceLimit,
-          approvedAmount
+          sqrtPriceLimit
```
Comment (Sniezka1927) [inline src/invariant.ts:252]: If `approvedAmount` has been removed from `TxScript` is it still required to pass it as parameter?

File: src/invariant.ts [hunk 3]
```diff
@@ -180,8 +178,6 @@ export class Invariant {
           lowerTick,
           upperTick,
           liquidityDelta,
```
Comment (Sniezka1927) [inline src/invariant.ts:180]: Same as https://github.com/invariant-labs/protocol-alephium/pull/99#discussion_r1677444724

**Review by Sniezka1927 — CHANGES_REQUESTED** [review]

**Review by Sniezka1927 — APPROVED** [review]

File: src/invariant.ts (hunk 2 above)
Comment (DevRozaDev) [reply src/invariant.ts:252]: Currently yes, we need to pass both the amount we want to receive and the approved one. In TxScripts they are passed in braces syntax.

**Review by wojciech-cichocki — APPROVED** [review]

**Pull request merged**